mozilla_django_oidc/auth.py

```diff
@@ -68,12 +101,28 @@ def authenticate(self, code=None, state=None):
             user_response = requests.get('{url}?{query}'.format(url=self.OIDC_OP_USER_ENDPOINT,
                                                                 query=query))
             user_info = user_response.json()
+            email = user_info.get('email')
+            if email is None:
+                return None
 
             try:
-                return self.UserModel.objects.get(email=user_info['email'])
+                return self.UserModel.objects.get(email=email)
+            except self.UserModel.MultipleObjectsReturned:
+                # In the rare case that two user accounts have the same email address,
+                # log and bail. Randomly selecting one seems really wrong.
+                LOGGER.warn('Multiple users with email address %s.', email)
+                return None
             except self.UserModel.DoesNotExist:
-                return self.UserModel.objects.create_user(username=user_info['nickname'],
-                                                          email=user_info['email'])
+                pass
```

Comment:
I really don't like the idea of having a block of code that does unrelated to the exception stuff inside the `except` clause. If the program flow has reached this point, all we have to do is decide upon the user creation, thus the `pass` here.